### DISCUSSION TITLE
Best practices: .chezmoi.sourceDir vs absolute path?

### DISCUSSION THREAD

**Discussion by HelloGrayson** [twpayne/chezmoi#3467] [General]
What's the case for using `.chezmoi.sourceDir` in templates and scripts vs `~/.local/share/chezmoi`?

In my dotfiles project I've been using the [following pattern](https://github.com/HelloGrayson/workstation/blob/main/src/backup/snapshot.sh#L5) in bash scripts:

```bash
WORKINGDIR="$(chezmoi data | jq .chezmoi.sourceDir -r)/src/backup"

bash "$WORKINGDIR/init-systemd.sh"
```

This grants scripts reliable access to eachother during a Chezmoi run.

Is there some advantage to this as opposed to the following convention?

```bash
WORKINGDIR="$HOME/.local/share/chezmoi/src/backup"

bash "$WORKINGDIR/init-systemd.sh"
```

For example, under what circumstances can the `sourceDir` change, or is that a constant?

Thanks for the feedback.

**Comment by twpayne** [comment]
If you're always going to use `~/.local/share/chezmoi` as your source directory you can use the fixed path.

Since some people want to use different directories (e.g. `~/dotfiles` or `~/repos/dots` or whatever), many examples in chezmoi's documentation use the `.chezmoi.sourceDir` template variable or `chezmoi source-path` command for portability. These also account for the user's [`.chezmoiroot`](https://www.chezmoi.io/reference/special-files-and-directories/chezmoiroot/) setting.

**Reply by HelloGrayson** [reply]
That explains it. 

So ultimately users can make their chezmoi source path portable if needed. In that case I think I will use the absolute path, simplifying my scripts, until such a need arrives.

Thank you, @twpayne.
